python_modules/dagster/dagster/core/host_representation/repository_location.py

```diff
@@ -180,6 +182,10 @@ def get_external_sensor_execution_data(
     ) -> Union["SensorExecutionData", "ExternalSensorExecutionErrorData"]:
         pass
 
+    @abstractmethod
+    def get_external_notebook_data(self, notebook_path: str) -> bytes:
```

Comment:
this will be a breaking change in the internal repo (porting this into the internal repo would be great though, if not urgent)

Reply:
@gibsondan can i get another pass on this PR? 

i wanted to make sure that i'm making the necessary threading on this PR for https://github.com/dagster-io/internal/pull/128: in order to get the right client, we will need to thread through the `repository_handle` even if it's not being used for accessing the notebook file in this case. so here on this pr, i need to add `repository_handle` arg to all `get_external_notebook_data` for the cloud api to figure out the right client to call (?)